app/models/miq_worker.rb

```diff
@@ -251,18 +252,22 @@ def self.log_status(level = :info)
     find_current.each { |w| w.log_status(level) }
   end
 
-  def self.create_worker_record(*params)
+  def self.init_worker_object(*params)
     params                  = params.first
     params                  = {} unless params.kind_of?(Hash)
     params[:queue_name]     = default_queue_name unless params.key?(:queue_name) || default_queue_name.nil?
     params[:status]         = STATUS_CREATING
     params[:last_heartbeat] = Time.now.utc
 
-    server_scope.create(params)
+    server_scope.new(params)
+  end
+
+  def self.create_worker_record(*params)
+    init_worker_object(*params).tap(&:save)
   end
 
   def self.start_worker(*params)
-    w = create_worker_record(*params)
+    w = containerized_worker? ? init_worker_object(*params) : create_worker_record(*params)
```

Comment:
Okay, I think I see what your doing here, and why, but curious what this means for the future:

* To clarify, we are doing creating worker records because it is too hard to manage our DB records with the containers that are currently running on OpenShift?
* Does this mean scripts/tools like `evm:status` would have to be completely re-tooled to work when running in a container env?
* Is the plan just to eventually drop these tables?
* Does this mess any other portions of the `MiqServer#monitor_loop`, now that there aren't DB records?  (looks like you covered `stop_worker`, but curious if there are other pieces I am missing)

Reply:
To be clear, we're still going to have worker records, it's just that they need to be [created by the worker container](https://github.com/ManageIQ/manageiq/blob/master/lib/workers/bin/run_single_worker.rb#L77) rather than by the server.

This is really because of the worker guid. We can't generate the guid here because we can't tell a replica what guid it should be using. The solution is to have the worker itself create the guid, and by extension the entire record.

Reply:
> it's just that they need to be created by the worker container rather than by the server.

Doh... I even wrote the code that does that and forgot... okay, guess that pretty much answers most of the questions then huh?

> We can't generate the guid here because we can't tell a replica what guid it should be using.

Forgive my lacking of openshift knowledge, but I assume we can't even pass a replica a ENV var if we were to make a change to support that in the `run_single_worker.rb` script as well?  I assume no, but just thought I would check and make sure.

Reply:
> I assume we can't even pass a replica a ENV var if we were to make a change to support that in the run_single_worker.rb script as well?

Correct, the environment for the container is defined at the deployment config level so each replica will have the same set of environment variables.